x-pack/metricbeat/module/aws/rds/rds.go

```diff
@@ -0,0 +1,256 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package rds
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/rds/rdsiface"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/rds"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/metricbeat/module/aws"
+)
+
+var metricsetName = "rds"
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(aws.ModuleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	*aws.MetricSet
+}
+
+// DBDetails holds detailed information from DescribeDBInstances for each rds.
+type DBDetails struct {
+	dbClass            string
+	dbAvailabilityZone string
+	dbIdentifier       string
+	dbStatus           string
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	metricSet, err := aws.NewMetricSet(base)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating aws metricset")
+	}
+
+	// Check if period is set to be multiple of 60s
+	remainder := metricSet.PeriodInSec % 60
+	if remainder != 0 {
+		err := errors.New("Period needs to be set to 60s (or a multiple of 60s). To avoid data missing or " +
+			"extra costs, please make sure period is set correctly in config.yml")
+		base.Logger().Info(err)
+	}
+
+	return &MetricSet{
+		MetricSet: metricSet,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) error {
+	// Get startTime and endTime
+	startTime, endTime, err := aws.GetStartTimeEndTime(m.DurationString)
+	if err != nil {
+		err = errors.Wrap(err, "Error ParseDuration")
+		report.Error(err)
+		return err
```

Comment:
New you report the error twice. You can redirectly return the error.

Reply:
oh no @fearful-symmetry I totally misled you...